wallet/src/de/schildbach/wallet/ui/dashpay/PlatformRepo.kt

```diff
@@ -180,14 +187,13 @@ class PlatformRepo private constructor(val walletApplication: WalletApplication)
      * @param text The beginning of a username to search for
      * @return
      */
+
     @Throws(Exception::class)
     suspend fun searchUsernames(text: String, onlyExactUsername: Boolean = false): List<UsernameSearchResult> {
         val wallet = walletApplication.wallet
         val blockchainIdentityData = blockchainIdentityDataDao.load()!!
-        //We don't check for nullity here because if it's null, it'll be thrown, captured below
-        //and sent as a Resource.error
-        val creditFundingTx = wallet.getCreditFundingTransaction(wallet.getTransaction(blockchainIdentityData.creditFundingTxId))
-        val userId = creditFundingTx.creditBurnIdentityIdentifier.toStringBase58()
+        val userId = blockchainIdentity.uniqueIdString
```

Comment:
since the userId is saved in the database, we don't need to load it from the wallet.